specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -42,16 +43,52 @@ export function Notifications({
 
   React.useEffect(() => {
     let pullInterval = INITIAL_INTERVAL;
+    let lastFetchedTimestamp: Date | undefined;
+
     const handler = (): void => {
-      if (timeout !== undefined) globalThis.clearTimeout(timeout);
+      if (timeout !== undefined) {
+        globalThis.clearTimeout(timeout);
+      }
       pullInterval = INITIAL_INTERVAL;
-      if (document.visibilityState === 'visible') doFetch();
+      if (document.visibilityState === 'visible') {
+        doFetch();
+      }
     };
+
     document.addEventListener('visibilitychange', handler);
 
     let timeout: NodeJS.Timeout | undefined = undefined;
 
-    function doFetch(): void {
+    function doFetch(since?: Date): void {
+      const startFetchTimestamp = new Date();
+
+      const now = new Date();
+      const year = now.getFullYear();
+      const month = now.getMonth().toString();
+      const day = now.getDate().toString();
+      const hours = now.getHours().toString();
+      const minutes = now.getMinutes().toString();
+      const seconds = now.getSeconds().toString();
+
+      const formattedDate = `${year}-${month}-${day} ${hours}:${minutes}:${seconds}`;
+
+      const sinceDate = since;
+      const sinceYear = sinceDate?.getFullYear();
+      const sinceMonth = sinceDate?.getMonth().toString();
+      const sinceDay = sinceDate?.getDate().toString();
+      const sinceHours = sinceDate?.getHours().toString();
+      const sinceMinutes = sinceDate?.getMinutes().toString();
+      const sinceSeconds = sinceDate?.getSeconds().toString();
+
+      const formattedSince = `${sinceYear}-${sinceMonth}-${sinceDay} ${sinceHours}:${sinceMinutes}:${sinceSeconds}`;
+
+      const queryString = since ? formattedSince : formattedDate;
```

Comment:
improve variable name. this variable is not a query string.
a query string looks like `?name=value` - this isn't that